ecs/execution-role.tf

```diff
@@ -146,11 +146,18 @@ resource "aws_iam_role_policy" "ecs-task-execution-policy" {
   })
 }
 
+locals {
+  kms_and_secret_arns = flatten([
+    var.secrets_arns,
+    var.kms_key_arns,
+  ])
+}
+
 resource "aws_iam_role_policy" "ecs-task-execution-secrets-policy" {
   name = "ecs-task-execution-secrets-policy"
   role = aws_iam_role.ecs-task-execution.name
 
-  policy = jsonencode({
+  policy = length(local.kms_and_secret_arns) > 0 ? jsonencode({
```

Comment:
I think we have to use count instead of making the policy an empty string because it will throw error below:
```
"policy" contains an invalid JSON policy
```

Reply:
I wanted to avoid count, since that would change the resource name in the terraform state, and thus it can break existing implementations :(

Reply:
Can we add a comment to explain why we have the dummy policy?

Reply:
done! :)